src/modules/dashboard/components/Dialogs/EscrowFundsDialog/EscrowFundsDialogForm.tsx

```diff
@@ -0,0 +1,384 @@
+  },
+  confirmText: {
+    id: 'dashboard.EscrowFundsDialog.EscrowFundsDialogForm.confirmText',
+    defaultMessage: 'Create motion',
+  },
+  fullFund: {
+    id: 'dashboard.EscrowFundsDialog.EscrowFundsDialogForm.fullFund',
+    defaultMessage: 'Total {name} to fund',
+  },
+  partialFund: {
+    id: 'dashboard.EscrowFundsDialog.EscrowFundsDialogForm.partialFund',
+    defaultMessage: '{name} to fund',
+  },
+  total: {
+    id: 'dashboard.EscrowFundsDialog.EscrowFundsDialogForm.total',
+    defaultMessage: 'Total',
+  },
+  createDomain: {
+    id: 'dashboard.EscrowFundsDialog.EscrowFundsDialogForm.createDomain',
+    defaultMessage: 'Motion will be created in',
+  },
+});
+
+interface Props {
+  colony: Colony;
+  onSubmitClick: () => void;
+  isVotingExtensionEnabled: boolean;
+  cancel: VoidFunction;
+}
+
+const EscrowFundsDialogForm = ({
+  colony,
+  onSubmitClick,
+  isVotingExtensionEnabled,
+  values,
+  handleSubmit,
+  isSubmitting,
+  cancel,
+}: Props & FormikProps<FormValues>) => {
+  const [domainID, setDomainID] = useState<number>();
+  const { formatMessage } = useIntl();
+  const { walletAddress, username, ethereal } = useLoggedInUser();
+  const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
+
+  const hasRegisteredProfile = !!username && !ethereal;
+  const canCancelExpenditure = hasRegisteredProfile && hasRoot(allUserRoles);
+
+  const [userHasPermission] = useDialogActionPermissions(
+    colony.colonyAddress,
+    canCancelExpenditure,
+    isVotingExtensionEnabled,
+    values.forceAction,
+  );
+
+  const handleMotionDomainChange = useCallback(
+    (motionDomainId) => setDomainID(motionDomainId),
+    [],
+  );
+
+  const getDomainColor = useCallback<(domainId: string | undefined) => Color>(
+    (domainId) => {
+      const rootDomainColor: Color = Color.LightPink;
+      const defaultColor: Color = Color.Yellow;
+      if (domainId === String(ROOT_DOMAIN_ID)) {
+        return rootDomainColor;
+      }
+      if (!colony || !domainId) {
+        return defaultColor;
+      }
+      const domain = colony?.domains.find(
+        ({ ethDomainId }) => Number(domainId) === ethDomainId,
+      );
+      return domain ? domain.color : defaultColor;
+    },
+    [colony],
+  );
+
+  const renderActiveOption = useCallback<
+    (option: SelectOption | undefined, label: string) => ReactNode
+  >(
+    (option, label) => {
+      const value = option?.value;
+      const color = getDomainColor(value);
+
+      return (
+        <div className={styles.activeItem}>
+          <ColorTag color={color} />
+          <div className={styles.activeItemLabel}>{label}</div>
+        </div>
+      );
+    },
+    [getDomainColor],
+  );
+
+  const filterDomains = useCallback((optionDomain) => {
+    const optionDomainId = parseInt(optionDomain.value, 10);
+
+    return optionDomainId !== COLONY_TOTAL_BALANCE_DOMAIN_ID;
+  }, []);
+
+  return (
+    <Dialog cancel={cancel}>
```

Comment:
I'd prefer for the Dialog component to be moved one level above - just for consistency with all of our other dialogs/forms